src/graph/test/DataTest.cpp

```diff
@@ -0,0 +1,291 @@
+        }
+    }
+    sleep(FLAGS_load_data_interval_secs + 1);
+    return TestOK();
+}
+
+AssertionResult DataTest::removeData() {
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "DROP SPACE mySpace";
+        auto code = client_->execute(cmd, resp);
+        if (cpp2::ErrorCode::SUCCEEDED != code) {
+            return TestError() << "Do cmd:" << cmd << " failed";
+        }
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = folly::stringPrintf("REMOVE HOSTS 127.0.0.1:%u",
+                                              gEnv->storageServerPort());
+        auto code = client_->execute(cmd, resp);
+        if (cpp2::ErrorCode::SUCCEEDED != code) {
+            return TestError() << "Do cmd:" << cmd << " failed";
+        }
+    }
+
+    return TestOK();
+}
+
+TEST_F(DataTest, InsertVertex) {
+    // Insert wrong type value
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT VERTEX person(name, age) VALUES 1001:(\"peiqi\", \"2\")";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_NE(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    // Insert wrong num of value
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT VERTEX person(name) VALUES 1001:(\"peiqi\", 2)";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_NE(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    // Insert wrong field
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT VERTEX person(Name, age) VALUES 1001:(\"peiqi\", 3)";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_NE(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT VERTEX person(name, age) VALUES 1001:(\"zhangsan\", 22)";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT VERTEX person(name, age) VALUES 1002:(\"lisi\", 23)";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    // One vertex multi tags
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT VERTEX person(name, age),student(grade, number) "
+                          "VALUES 1003:(\"xiaoming\", 8, \"three\", 20190901001)";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    // Multi vertices multi tags
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT VERTEX person(name, age),student(grade, number) "
+                          "VALUES 1005:(\"xiaoqiang\", 8, \"three\", 20190901008),"
+                          "1006:(\"xiaohong\", 9, \"four\", 20180901003)";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+    }
```

Comment:
insert vertex person(xx,xx) values 111(xxx); insert vertex student(xx,xx) values 111(xxxx);  -- two separate insertions in one statement.
and test in parser unittest.

Reply:
good advice